src/preact/parse-props.js

```diff
@@ -312,11 +309,15 @@ function matchChild(element, defs) {
 /**
  * @param {string} name
  * @param {function(string): T} parse
- * @return {{attrs: Array<string>, parseAttrs: function(!Element):(?T|undefined)}}
+ * @return {{
+ *   attrs: Array<string>,
+ *   parseAttrs: function(!Element):(T|''|null)
+ * }}
  * @template T
  */
 export function createParseAttr(name, parse) {
   const attrs = [name];
+  /** @type {function(Element):null|T|''} */
```

Comment:
nit: match return type order from above
```suggestion
  /** @type {function(Element):T|''|null} */
```

Reply:
Done!